### PR TITLE
64bit and fixes

### PR DESCRIPTION
**Pull request by miguelsousa**
This is the revival of PR #161

### CODE REVIEW

**Comment by miguelsousa** [comment]
@readroberts why re-indenting whole files when fixing the affected lines would have been simpler and would not introduce a bunch of unnecessary changes?

This PR is messed up too much for my taste.

**Comment by readroberts** [comment]
@msousa  I choose to re-indent the entire file because there are many  inconsistencies in indentation and spacing throughout the file. If I am fixing one spot, why not make them all consistent? I note that the only changes in the last commit are whitespace.  I don't care that much about the issue - if you really don't want the entire file re-indented, I can revert the change, let me know, and I will fix only the spots that recently had tabs converted to spaces by Xcode.

**Comment by miguelsousa** [comment]
@readroberts changing just the lines I marked would have been a lot better IMO. Please undo the changes by rewriting the commits instead of reverting.

**Comment by cjchapman** [comment]
@miguelsousa & @readroberts There are some easy-to-fix issues in the Codacy report, having to do with using "%ld" in `printf` formatting for integers that are not `long`... just need to change them to "%d". (This does not constitute a review, just a quick observation, I'll do a review later.)

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/cffread/cffread.c [hunk 1]
```diff
@@ -86,7 +86,7 @@ typedef struct {                  /* FDArray element */
 typedef union {                   /* Stack element */
 	double d;
 	Fixed f;
-	long l;
+	int32_t l;
```
Comment (cjchapman) [inline afdko/Tools/Programs/makeotf/makeotf_lib/source/cffread/cffread.c]: The `l` was a `long`, but is now an `int32_t`, perhaps the name should be changed to `i` to reflect this?
(see related comments below)

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/GPOS.c [hunk 2]
```diff
@@ -4764,7 +4755,7 @@ static void writeExtension(hotCtx g, GPOSCtx h, Subtable *sub) {
 	/* Adjust offset */
 	fmt->ExtensionOffset += h->offset.extensionSection - sub->offset;
 
-	DF(1, (stderr, "  GPOS Extension: fmt=%1d, lkpType=%2d, offset=%08lx\n",
+	DF(1, (stderr, "  GPOS Extension: fmt=%1d, lkpType=%2d, offset=%08ux\n",
```
[the remark this answers is not in the document]
Comment (cjchapman) [reply afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/GPOS.c]: @readroberts, to fix this issue, I think you just need to change this:
```C
#define DF_LEVEL (g->font.debug & HOT_DB_FEAT_2 ? 2 : \
                  (g->font.debug & HOT_DB_FEAT_1 ? 1 : 0))
```
to:
```C
#define DF_LEVEL ((g->font.debug & HOT_DB_FEAT_2) ? 2 : \
                  ((g->font.debug & HOT_DB_FEAT_1) ? 1 : 0))
```
in `feat.h`.

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/typecomp/tc.c [hunk 3]
```diff
@@ -606,7 +606,7 @@ static void showStats(tcprivCtx h) {
 		origOther = g->stats.fontSize -
 		    (g->stats.subrSize + g->stats.charSize);
 		compOther = h->FontSet.size - (comp.subrSize + comp.charSize);
-		printf("other@ %8ld %9ld %8ld  %8ld %9ld %8ld %6.1f%%\n",
```
[the remark this answers is not in the document]
Comment (miguelsousa) [reply afdko/Tools/Programs/makeotf/makeotf_lib/source/typecomp/tc.c]: @readroberts seems like you somehow redid and/or undid the fixes I committed in 37d19a2d3fb55f4435700495dffbfc95ef8212a8 :confused:

**Comment by readroberts** [comment]
I have now installed cppCheck - required fixing some stuff with Xcode. I recently installed High Sierra, but I had some old stuff that interfered with Xcode 9. cppCheck shows many warnings about makeotf - over 500, with '--enable=warning --enable=style'  - but this commit fixes all the issues with the change from 32 to 64 bit. I propose that this PR should be merged  with main branch now, and that the many cppCheck non-consequential style and printf format warnings should be pursued as a separate project.

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/GPOS.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (readroberts) [reply afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/GPOS.c]: cppCheck did complain about precedence ambiguity in the original version of the DF and DL_LEVEL macros, but does not complain now about these lines, and the expanded DF_LEVEL and DF macro code looks correct to me. I don't see a problem to fix here.

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/cffread/cffread.c [hunk 4]
```diff
@@ -34,7 +34,7 @@ static short exenc[] = {
 #define ABS(v)            ((v) < 0 ? -(v) : (v))
 
 /* 16.16 fixed point arithmetic */
-typedef long Fixed;
+typedef int32_t Fixed;
```
[the remark this answers is not in the document]
Comment (miguelsousa) [reply afdko/Tools/Programs/makeotf/makeotf_lib/source/cffread/cffread.c]: It's done. Line 2759 and below

File: afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/map.c [hunk 5]
```diff
@@ -2540,10 +2540,10 @@ static void dbgUniBlock(hotCtx g) {
...
 static void dbgPrintUV(UV uv) {
 	if (IS_SUPP_UV(uv)) {
-		fprintf(stderr, "%lX", uv);
+		fprintf(stderr, "%uX", uv);
 	}
 	else {
-		fprintf(stderr, "%04lX", uv);
+		fprintf(stderr, "%04uX", uv);
 	}
 }
```
[the remark this answers is not in the document]
Comment (miguelsousa) [reply afdko/Tools/Programs/makeotf/makeotf_lib/source/hotconv/map.c]: This is done

**Pull request merged**